### PR TITLE
Ndarray wave

### PR DESCRIPTION
**Pull request by brelod**
Found the way to inherit from `np.ndarray` (https://numpy.org/doc/stable/user/basics.subclassing.html)
New unittests were added to verify the functionality of `Wave` class

**Example usage:**
```
a = Wave(range(10), fs=1)
b = Wave(range(5), fs=1)

# Addition:
c = a + b

# Shift:
b.shift(5)
b.shift(-5)

# Extend:
a.extend(b)
a.extend(b, left=True)

# Normalise:
a.norm()
b.norm(5) 
```

**Note**: Further tests should be written for Wave class

### CODE REVIEW

**Comment by brelod** [comment]
In the meantime I've found another solution.
In this one we are rather using a functional programming way (in my understanding) because we can not really extend the ndarray (memory has to be reallocated) so we are returning with a new array instead in the extend/shift methods.

On the other hand if we wanna do it in a more encapsulated way then we can create an ndarray holder protected property inside the wave and create all the necessary methods manually.

**Example**:
```
import matplotlib.pyplot as plt
import numpy as np

class Wave(list):
    def __init__(self, array, fs):
        self._array = np.array(array)
        self.fs = fs

    def __repr__(self):
        return "%s(%s)" % (self.__class__.__name__, self._array)

    def __iter__(self):
        # This is necessary for iterations like the one in plt.plot()
        return iter(self._array)

    def extend(self, array):
        self._array = np.concatenate((self._array, array))

    def __add__(self, other):
        return Wave(self._array + other._array, fs=1)

a = Wave([1, 2, 3], fs=1)
b = Wave([1, 2, 3], fs=1)


>>> a
Wave([1 2 3])
>>> b
Wave([1 2 3])
>>> a + b
Wave([2 4 6])
>>> a.extend(b)
>>> a
Wave([1 2 3 1 2 3])
>>> b
Wave([1 2 3])
>>> plt.plot(a)
[<matplotlib.lines.Line2D object at 0x7fc443468a90>]
>>> plt.show()
>>>
```

File: audio/sine_sound.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import numpy as np
 import cmath
 from bisect import bisect_right
-import sounddevice as sd
+#import sounddevice as sd
```
Comment (Tockos) [inline audio/sine_sound.py]: Don't do this

File: audio/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+"""
+Audio package for learning about digital signal processing especially about
```
Comment (Tockos) [inline audio/__init__.py:2]: nice

File: audio/wave.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+
+import numpy as np
+
+class Wave(np.ndarray):
+    def __new__(cls, array, fs=1):
+        obj = np.asarray(array).view(cls)
+        obj.fs = fs
+        return obj
+
+    def __len__(self):
```
Comment (Tockos) [inline audio/wave.py:10]: isn't necessary as len is defined for ndarray

File: audio/wave.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        return self.size
+
```
Comment (Tockos) [inline audio/wave.py]: isn't necessary as len is defined for ndarray

**Review by Tockos — CHANGES_REQUESTED** [review]
Impressed

File: audio/__init__.py (hunk 2 above)
Comment (brelod) [reply audio/__init__.py:2]: cheers

**Pull request merged**